[bug] crash on errors in v1.5.0

Today I noticed that `go-arg` is crashing on errors in `v1.5.0`, which did not happen in `v1.4.3` (previous version).

Consider the following minimal reproduction example:
```go
package main

import "github.com/alexflint/go-arg"

var args struct {
	Foo []string `default:""`
}

func main() {
	arg.MustParse(&args)
}
```
**Important:** Note that the struct-tag is invalid because `default` is not supported for slices.

When the above MRE is run with `v1.5.0`:
```
$ go get github.com/alexflint/go-arg@v1.5.0
$ go run main.go
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x4a1cce]

goroutine 1 [running]:
fmt.Fprintln({0x0, 0x0}, {0xc00011be98, 0x1, 0x1})
        /opt/go/src/fmt/print.go:305 +0x4e
github.com/alexflint/go-arg.mustParse({{0x0, 0x0}, 0x0, 0x0, 0x0, 0x4fa540, {0x0, 0x0}}, {0xc000059f30, 0x1, ...})
        /home/hhromic/go/pkg/mod/github.com/alexflint/go-arg@v1.5.0/parse.go:91 +0xe5
github.com/alexflint/go-arg.MustParse(...)
        /home/hhromic/go/pkg/mod/github.com/alexflint/go-arg@v1.5.0/parse.go:84
main.main()
        /home/hhromic/repro/main.go:10 +0x8e
exit status 2
```

However, when run with `v1.4.3`:
```
$ go get github.com/alexflint/go-arg@v1.4.3
go: downgraded github.com/alexflint/go-arg v1.5.0 => v1.4.3
$ go run main.go
.Foo: default values are not supported for slice or map fields
exit status 255
```
Then `go-arg` correctly reports the error.

The problem seems to be in `MustParse` and `mustParse` (the crash happens in line 91):
https://github.com/alexflint/go-arg/blob/dfca71d1594a749ef5e2203e84cfb5a1b8e9387b/parse.go#L82-L98

It can be seen that `MustParse` initializes the `Config` struct without `Out`, which remains as `nil`.

The initialization of `Out` seems to be done in `NewParser`:
https://github.com/alexflint/go-arg/blob/dfca71d1594a749ef5e2203e84cfb5a1b8e9387b/parse.go#L198-L207

However, the `config` argument is NOT a pointer, and therefore all changes made to it by `NewParser` remain local.

--- a/example_test.go
+++ b/example_test.go
@@ -163,6 +163,7 @@ func Example_helpText() {
 
 	// This is only necessary when running inside golang's runnable example harness
 	mustParseExit = func(int) {}
+	mustParseOut = os.Stdout
 
 	MustParse(&args)
 
@@ -195,6 +196,7 @@ func Example_helpPlaceholder() {
 
 	// This is only necessary when running inside golang's runnable example harness
 	mustParseExit = func(int) {}
+	mustParseOut = os.Stdout
 
 	MustParse(&args)
 
@@ -235,6 +237,7 @@ func Example_helpTextWithSubcommand() {
 
 	// This is only necessary when running inside golang's runnable example harness
 	mustParseExit = func(int) {}
+	mustParseOut = os.Stdout
 
 	MustParse(&args)
 
@@ -272,6 +275,7 @@ func Example_helpTextWhenUsingSubcommand() {
 
 	// This is only necessary when running inside golang's runnable example harness
 	mustParseExit = func(int) {}
+	mustParseOut = os.Stdout
 
 	MustParse(&args)
 
@@ -392,6 +396,7 @@ func Example_errorText() {
 
 	// This is only necessary when running inside golang's runnable example harness
 	mustParseExit = func(int) {}
+	mustParseOut = os.Stdout
 
 	MustParse(&args)
 
@@ -415,6 +420,7 @@ func Example_errorTextForSubcommand() {
 
 	// This is only necessary when running inside golang's runnable example harness
 	mustParseExit = func(int) {}
+	mustParseOut = os.Stdout
 
 	MustParse(&args)
 
@@ -450,6 +456,7 @@ func Example_subcommand() {
 
 	// This is only necessary when running inside golang's runnable example harness
 	mustParseExit = func(int) {}
+	mustParseOut = os.Stdout
 
 	MustParse(&args)
 

--- a/parse.go
+++ b/parse.go
@@ -76,12 +76,13 @@ var ErrHelp = errors.New("help requested by user")
 // ErrVersion indicates that the builtin --version was provided
 var ErrVersion = errors.New("version requested by user")
 
-// for monkey patching in example code
+// for monkey patching in example and test code
 var mustParseExit = os.Exit
+var mustParseOut io.Writer = os.Stdout
 
 // MustParse processes command line arguments and exits upon failure
 func MustParse(dest ...interface{}) *Parser {
-	return mustParse(Config{Exit: mustParseExit}, dest...)
+	return mustParse(Config{Exit: mustParseExit, Out: mustParseOut}, dest...)
 }
 
 // mustParse is a helper that facilitates testing

--- a/parse_test.go
+++ b/parse_test.go
@@ -692,6 +692,21 @@ func TestMustParse(t *testing.T) {
 	assert.NotNil(t, parser)
 }
 
+func TestMustParseError(t *testing.T) {
+	var args struct {
+		Foo []string `default:""`
+	}
+	var exitCode int
+	var stdout bytes.Buffer
+	mustParseExit = func(code int) { exitCode = code }
+	mustParseOut = &stdout
+	os.Args = []string{"example"}
+	parser := MustParse(&args)
+	assert.Nil(t, parser)
+	assert.Equal(t, -1, exitCode)
+	assert.Contains(t, stdout.String(), "default values are not supported for slice or map fields")
+}
+
 func TestEnvironmentVariable(t *testing.T) {
 	var args struct {
 		Foo string `arg:"env"`

